htdocs/bom/class/bom.class.php

```diff
@@ -396,6 +396,18 @@ public function fetchLines()
 		$this->lines = array();
 
 		$result = $this->fetchLinesCommon();
+
+		//associate to each line childbom if exist
+		if (!empty($this->lines)) {
+			foreach ($this->lines as $line) {
+				if($line->fk_bom_child > 0) {
+					$bom_child = new BOM($this->db);
+					$res = $bom_child->fetch($line->fk_bom_child);
```

Comment:
There is a risk of infinite loop if in database a bom A has child B that has child A...
Do you really need have a subrequest for child to load "immediatly" the child.
Can't you load it only when "you need it" ?